src/app/qgisapp.cpp

```diff
@@ -17184,12 +17184,6 @@ void QgisApp::handleRenderedLayerStatistics() const
       QgsMeshLayer *meshLayer = qobject_cast<QgsMeshLayer *>( QgsProject::instance()->mapLayer( layerStatistics->layerId() ) );
       if ( meshLayer )
       {
-        QgsMeshRendererSettings rendererSettings = meshLayer->rendererSettings();
```

Comment:
Where is this logic applied now?

Reply:
It is applied in step of renderer creation. Basically, this part of the logic was applied twice, which caused issue (double rendering). 

There is significant difference between mesh and raster renderer - mesh renderer is not stored at all, only the settings and mesh renderer is always created when it is needed. Thus there is no `refresh` of it.

I am not 100% sure that I properly understand the approach that was introduced with `QgsRenderedLayerStatistics`. Should the layer statistics be only applied on the renderer from this part of the code? And if rendered from anywhere else, just stick with data that are already set in mesh render settings?